docs/api/utils/mergeWithKey.md

```diff
@@ -1,17 +1,40 @@
 ---
-title: mergeWithKey()
+title: mergeWithKey(inputObject)
+sidebar_label: mergeWithKey()
 ---
 
 Emits the values from all the streams of the provided object, in a result
 which provides the key of the stream of that emission.
 
-### Arguments
+```ts
+function mergeWithKey<
+  O extends { [P in keyof any]: ObservableInput<any> },
+  OT extends {
+    [K in keyof O]: O[K] extends ObservableInput<infer V>
+      ? { type: K; payload: V }
+      : unknown
+  }
+>(inputObject: O, concurrent?: number, scheduler?: SchedulerLike): 
+  Observable<OT[keyof O]>
+```
+
+#### Arguments
 
 - `inputObject`: Object of streams
+- `concurrent`: (Optional) **TODO**
+- `scheduler`: (Optional) **TODO**
```

Comment:
The thing is that `mergeWithKey` is sugar for a common pattern `merge`, for instance this:

```ts
const todoActions$ = mergeWithKey({                   
  add: newTodo$.pipe(map(text, id) => ({ id, text })),
  edit: editTodo$,                                    
  toggle: toggleTodo$.pipe(map(id => ({ id }))),      
  delete: deleteTodo$.pipe(map(id => ({ id })))       
})                                                    
```                                                   

is equivalent to:

```ts
const todoActions$ = merge(        
  newTodo$.pipe(map(text, id) => ({
    type: "add" as const           
    payload: { id, text },         
  })),                             
  editTodo$.pipe(map(payload => ({ 
    type: "edit" as const,         
    payload,                       
  }))),                            
  toggleTodo$.pipe(map(id => ({    
    type: "toggle" as const,       
    payload: { id },               
  }))),                            
  deleteTodo$.pipe(map(id => ({    
    type: "delete" as const,       
    payload: { id },               
  }))),                            
)                                  
```

And since [the official `merge` operator takes those 2 optional arguments](https://github.com/ReactiveX/rxjs/blob/6.5.5/src/internal/observable/merge.ts#L112-L115), then `mergeWithKey` also accepts them and it propagates them to the underlying `merge`.

The official docs [don't explain very well](https://rxjs-dev.firebaseapp.com/api/index/function/merge) what those 2 optional arguments are for, but the comments on the source do explain it a bit better:

```
@param {number} [concurrent=Number.POSITIVE_INFINITY] Maximum number of input
Observables being subscribed to concurrently.

@param {SchedulerLike} [scheduler=null] The {@link SchedulerLike} to use for managing
concurrency of input Observables.
```

Reply:
This explanation should go in the docs! Will see if I can add it!